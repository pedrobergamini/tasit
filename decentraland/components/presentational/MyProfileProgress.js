import {
  responsiveHeight,
  responsiveWidth,
} from "react-native-responsive-dimensions";
import { StyleSheet, Text, View } from "react-native";
import React from "react";
import Colors from "@shared-constants/Colors";
import ProgressBar from "react-native-progress/Bar";
import PropTypes from "prop-types";

export function MyProfileProgress({ progress }) {
  return (
    <View style={styles.container}>
      <ProgressBar
        progress={progress}
        color={Colors.loadingColor}
        borderWidth={0}
        unfilledColor={Colors.textColor}
        height={responsiveHeight(1)}
        width={responsiveWidth(100)}
        borderRadius={0}
      />
      <MyProfileProgressText progress={progress} />
    </View>
  );
}

MyProfileProgress.propTypes = {
  progress: PropTypes.number,
};

export function MyProfileProgressText({ progress }) {
  const formattedProgress = Math.round(progress * 100);

  return (
    <View style={styles.progressTextContainer}>
      <Text style={styles.progressText}>{formattedProgress}% complete</Text>
    </View>
  );
}

MyProfileProgressText.propTypes = {
  progress: PropTypes.number,
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: Colors.backgroundColor,
    alignItems: "center",
    justifyContent: "flex-start",
  },
  progressTextContainer: {
    flexDirection: "row",
    alignItems: "center",
    marginTop: responsiveHeight(1),
  },
  progressText: {
    fontWeight: "800",
    color: Colors.textColor,
  },
});

export default MyProfileProgress;
