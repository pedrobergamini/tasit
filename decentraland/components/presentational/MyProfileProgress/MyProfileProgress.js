import {
  responsiveHeight,
  responsiveWidth,
} from "react-native-responsive-dimensions";
import { StyleSheet, Text, View } from "react-native";
import React from "react";
import Colors from "@shared-constants/Colors";
import ProgressBar from "react-native-progress/Bar";
import PropTypes from "prop-types";

export function MyProfileProgress({ progress }) {
  return (
    <View style={styles.container}>
      <ProgressBar
        borderRadius={0}
        borderWidth={0}
        color={Colors.loadingColor}
        height={responsiveHeight(1)}
        progress={progress}
        unfilledColor={Colors.textColor}
        width={responsiveWidth(100)}
      />
      <MyProfileProgressText progress={progress} />
    </View>
  );
}

MyProfileProgress.propTypes = {
  progress: PropTypes.number,
};

export function MyProfileProgressText({ progress }) {
  const formattedProgress = Math.round(progress * 100);

  return (
    <View style={styles.progressTextContainer}>
      <Text style={styles.progressText}>{formattedProgress}% complete</Text>
    </View>
  );
}

MyProfileProgressText.propTypes = {
  progress: PropTypes.number,
};

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    backgroundColor: Colors.backgroundColor,
    flex: 1,
    flexDirection: "column",
    justifyContent: "flex-start",
  },
  progressText: {
    color: Colors.textColor,
    fontWeight: "800",
  },
  progressTextContainer: {
    alignItems: "center",
    flexDirection: "row",
    marginTop: responsiveHeight(1),
  },
});

export default MyProfileProgress;
