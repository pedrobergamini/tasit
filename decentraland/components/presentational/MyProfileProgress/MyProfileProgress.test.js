import React from "react";
import { shallow } from "enzyme";
import { MyProfileProgress, MyProfileProgressText } from "./MyProfileProgress";

// Not the @presentational version since it needs
// the internal component

describe("MyProfileProgress", () => {
  it("renders the component", async () => {
    const progress = 0.5;
    expect(
      shallow(<MyProfileProgress progress={progress} />)
    ).toMatchSnapshot();
  });

  describe("MyAccountProgressText", () => {
    it("renders the component", async () => {
      const progress = 0.5;
      expect(
        shallow(<MyProfileProgressText progress={progress} />)
      ).toMatchSnapshot();
    });
  });
});
