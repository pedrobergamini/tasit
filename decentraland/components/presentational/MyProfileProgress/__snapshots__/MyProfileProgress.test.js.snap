// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfileProgress MyAccountProgressText renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginTop": 13.34,
    }
  }
>
  <Component
    style={
      Object {
        "color": "#CCCCCC",
        "fontWeight": "800",
      }
    }
  >
    50
    % complete
  </Component>
</Component>
`;

exports[`MyProfileProgress renders the component 1`] = `
<Component
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={0}
    borderWidth={0}
    color="#07FFC2"
    height={13.34}
    indeterminate={false}
    indeterminateAnimationDuration={1000}
    progress={0.5}
    unfilledColor="#CCCCCC"
    useNativeDriver={false}
    width={750}
  />
  <MyProfileProgressText
    progress={0.5}
  />
</Component>
`;
